### PR TITLE
Updates unmaintained tokio-retry to tokio-retry2

### PR DESCRIPTION
**Pull request by naomijub**
[`tokio-retry`](https://crates.io/crates/tokio-retry) has been unmaintained for over 3 years now, so this new crate, [tokio-retry2](https://crates.io/crates/tokio-retry2), is being used by my project. 

It has new features like the ability to early exit with wrapping errors (RetryError::Permanent), or continue the retry loop (`RetryError::Transient`), as well a notification on fail system (which was not included in this PR, but can be easily done replacing `Retry::spawn(..)` with `Retry::spawn_notify(..)` .

### CODE REVIEW

**Comment by naomijub** [comment]
Github Actions can be found here https://github.com/naomijub/redis-rs/pull/1

File: redis/src/aio/connection_manager.rs [hunk 1]
```diff
@@ -316,10 +315,13 @@ impl ConnectionManager {
         number_of_retries: usize,
         connection_config: &AsyncConnectionConfig,
     ) -> RedisResult<MultiplexedConnection> {
-        let retry_strategy = exponential_backoff.map(jitter).take(number_of_retries);
+        let retry_strategy = exponential_backoff.take(number_of_retries);
```
Comment (nihohit) [inline redis/src/aio/connection_manager.rs:318]: why did you remove the jitter?

Comment (naomijub) [reply redis/src/aio/connection_manager.rs:318]: No specific reason, for my team the jitter was an issue and we made it into an optional feature. I can add it back if was a desired behaviour

Comment (nihohit) [reply redis/src/aio/connection_manager.rs:318]: Please do.

Comment (naomijub) [reply redis/src/aio/connection_manager.rs]: [917ea10](https://github.com/redis-rs/redis-rs/pull/1334/commits/917ea1042b7fad9363210a4032ddd079c5efe18e)
